test/distributed/fsdp/test_fsdp_comm.py

```diff
@@ -0,0 +1,148 @@
+# Owner(s): ["oncall: distributed"]
+
+import sys
+from unittest.mock import patch
+
+import torch
+from torch import distributed as dist
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest, NestedWrappedModule
+from torch.testing._internal.common_utils import (
+    TEST_WITH_DEV_DBG_ASAN,
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+)
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestCommunication(FSDPTest):
+    """Tests ``FullyShardedDataParallel``'s collective communication usage."""
+    @skip_if_lt_x_gpu(2)
+    @parametrize(
```

Comment:
Good point. I will reduce the number of tests to 4 by always using `num_no_sync_iters = 3` and `num_sync_iters = 3`. I had both being either `1` or `3` since we originally suspected that there may be different behavior depending on how many iterations were run.